lib/pure/sugar.nim

```diff
@@ -411,10 +411,13 @@ macro collect*(body: untyped): untyped {.since: (1, 5).} =
       for i, d in data.pairs: {i: d}
     assert m == {0: "bird", 1: "word"}.toTable
 
-    # avoid `collect` when `sequtils.toSeq` suffices:
-    assert collect(for i in 1..3: i*i) == @[1, 4, 9] # ok in this case
```

Comment:
It's saying the `i*i` case is fine, not overkill. And honestly `mapIt` is probably not simpler in how it's implemented at all. `collect` is good because it's uniform, and it supports more than just seqs. 

If one wishes for a shorter syntax it's not hard to define a `collectIt`.

<details>

```nim
import sugar

template collectIt(coll, elem): untyped =
  collect(for it {.inject.} in coll: elem)

assert collectIt(1..3, it*it) == @[1, 4, 9]
```

</details>

Reply:
It's not only because it's shorter but also because it's more efficient. In this case the result seq has a known length (the length of the input iterable), `collect` `add`s while `mapIt` preallocates the correct length and fills with `[]=`.

Reply:
I forgot about that, you're right. It is possible to use `newSeqOfCap` with collect though to allocate the same amount of memory. Also neither case will be able to tell the length of iterator calls, but again, collect gives you more control with newSeqOfCap.

Reply:
`[]=` is still faster than `add` even with the right amount of memory allocated.

Reply:
Personally, I don't like these macros very much - I'd rather have `toSeq`, `toSet`, etc., take a variety of expressions (or `exprToSeq`, etc.). The correct routine to use should be obvious to the reader, based upon its name, and there should only be one obvious routine to use for any given scenario.

In any case, rather than give suggestions about performance and usage scenarios, I feel that this documentation would be better improved by simply reference toSeq, and providing a short comparison:
```
See also `toSeq` for more simplified use-cases.

Reply:
I somewhat don't like them either. Ig concepts (like `Iterable[T]`) should make things better.

Reply:
I added a `See also` section and removed the `runnableExamples`.